src/components/Radio/Radio.tsx

```diff
@@ -1,66 +1,157 @@
  */
-class Radio extends UIComponent<Extendable<IRadioProps>, any> {
+class Radio extends AutoControlledComponent<Extendable<IRadioProps>, any> {
+  static create: Function
+
   static displayName = 'Radio'
 
   static className = 'ui-radio'
 
   static propTypes = {
     as: customPropTypes.as,
 
+    /** Whether or not radio is checked. */
+    checked: PropTypes.bool,
+
     /** Child content * */
     children: PropTypes.node,
 
     /** Additional classes. */
     className: PropTypes.string,
 
+    /** Initial checked value. */
+    defaultChecked: PropTypes.bool,
+
+    /** A radio can appear disabled and be unable to change states. */
+    disabled: PropTypes.bool,
+
     /** The label of the radio input. */
     label: PropTypes.string,
 
+    /** The HTML input name. */
+    name: PropTypes.string,
+
+    /**
+     * Called after radio is clicked.
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     */
+    onChange: PropTypes.func,
+
     /** Custom styles to be applied for component. */
     styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
 
     /** The HTML input type. */
     type: PropTypes.string,
 
+    /** The HTML input value. */
+    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+
     /** Custom variables to be applied for component. */
     variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
   }
 
-  static handledProps = ['as', 'children', 'className', 'label', 'styles', 'type', 'variables']
+  static handledProps = [
+    'as',
+    'checked',
+    'children',
+    'className',
+    'defaultChecked',
+    'disabled',
+    'label',
+    'name',
+    'onChange',
+    'styles',
+    'type',
+    'value',
+    'variables',
+  ]
 
   static defaultProps = {
     as: 'div',
     type: 'radio',
   }
 
-  renderComponent({ ElementType, classes, rest, styles }) {
-    const { type, label } = this.props
+  static autoControlledProps = ['checked']
+
+  private handleChange = (e: React.SyntheticEvent) => {
+    const { onChange, disabled } = this.props
+    const checked = (e.target as HTMLInputElement).checked
+
+    if (disabled) {
+      e.preventDefault()
+      return
+    }
+
+    this.trySetState({ checked })
+
+    if (onChange) {
+      onChange(e, { ...this.props, checked })
+    }
+  }
+
+  renderComponent({ ElementType, classes, rest, styles, variables }) {
+    const { type, label, disabled, value, name, icon } = this.props
+    const { checked } = this.state
 
     return (
       <ElementType {...rest} className={classes.root}>
-        <Label styles={{ root: styles.label }}>
+        <Label as="label" styles={{ root: styles.label }}>
+          {Icon.create(icon || '', {
+            defaultProps: {
+              circular: true,
+              size: 'mini',
+              variables: {
+                color: checked ? variables.checkedIconColor : variables.uncheckedIconColor,
```

Comment:
I see you are already discussing this in #207. It would be great if, eventually, we would not need to touch variables in component code.

Reply:
unfortunately, this makes 'variables defined in theme' story a bit more complicated, as now we might need to get an access to component props there - as it is discussed in #207. This would be quite significant move, and we should consider whether there are any other not-so-radical alternatives. Still, totally agree that we should finally come up with solution that will eliminate this code from component's logic

Reply:
Exactly I was not satisfied with having to use the variable in the renderComponent, but as I mention there we should either be able to define the variables depending on the state and props, or rethink about the API provided so far for the components